src/api/v1/community_businesses/temporary/delete.ts

```diff
@@ -0,0 +1,54 @@
+import * as Hapi from 'hapi';
+import * as Boom from 'boom';
+import { getCommunityBusiness, isChildOrganisation } from '../../prerequisites';
+import { id, response } from '../schema';
+
+export default [
+  {
+    method: 'DELETE',
+    path: '/community-businesses/temporary/{organisationId}',
+    options: {
+      description: 'Hard delete a temporary community businesses and all associated data',
+      auth: {
+        strategy: 'standard',
+        access: {
+          scope: ['organisations_details-child:delete'],
+        },
+      },
+      validate: {
+        params: { organisationId: id },
+      },
+      response: { schema: response },
+      pre: [
+        { method: getCommunityBusiness, assign: 'communityBusiness' },
+        { method: isChildOrganisation, assign: 'isChild' },
+      ],
+    },
+    handler: async (request: Hapi.Request, h: Hapi.ResponseToolkit) => {
+      const { pre: { communityBusiness, isChild }, server: { app: { knex } } } = request;
+
+      if (!isChild) {
+        return Boom.forbidden('Insufficient permissions to access this resource');
+      }
+
+      const isTempAccount = /TEMPORARY ACCOUNT: \w+/.test(communityBusiness.name);
+      if (!isTempAccount) {
+        return Boom.forbidden('Not a temporary organisation');
+      }
+
+      await knex.transaction(async (trx) => {
+        // delete all users at this organisation
+        await trx.raw('DELETE FROM user_account '
+        + 'USING user_account_access_role '
+        + 'WHERE user_account.user_account_id = user_account_access_role.user_account_id '
+        + 'AND user_account_access_role.organisation_id = ?', [communityBusiness.id]);
```

Comment:
Does this need a `CASCADE`?

Reply:
How do you mean? I was under the impression `cascade` is covered by the table constraints not something implemented in queries.